### DISCUSSION TITLE
Access multiple stores with their name as string

### DISCUSSION THREAD

**Discussion by kerbilg** [vuejs/pinia#1743] [Help and Questions] [answered]
Hi, I am working on a project with multiple stores.

So in my project, I export all my stores in one single module for easy access. 

Because I am migrating an older vuex project, the easiest solution will be to access the stores with their name as a string.

The file looks like this: (@/stores/index.ts)
```
import { useStore1 } from './store1'
import { useStore2 } from './store2'
import { useStore3 } from './store3'

export {
  useStore1,
  useStore2,
  useStore3
}
```

I need to run an action, in which the action and store name are given in a variable called object. 
object.store and object.function are strings

```
import * as stores from '@/stores'

const tempStore = stores[object.store]()
tempStore[object.function]()
```

for stores[object.store] i get the error message:
```
Element implicitly has an 'any' type because the expression of type 'string' can't be used to index type 'typeof import("@/stores/index")'.
  No index signature with a parameter of type 'string' was found on type 'typeof import("@/stores/index")'.ts(7053)
```

Has anyone an idea what type I can use to access the store?

Thank you for your time!

**Comment by posva** [comment, marked as answer]
you need `object.store` to be of type`'useStore1' | 'useStore2' | 'useStore3'` (instead of just `string`) for TS to figure out the type
